### PR TITLE
bytes: clean up and simplify encoding of HexBytes

### PR DESCRIPTION
**Pull request by creachadair**
As written, the encoding step unnecessarily made and moved multiple copies of
the encoded representation. Reduce this to a single allocation and encode the
data in-place so that a shift is no longer required.

No functional changes.

### CODE REVIEW

File: libs/bytes/bytes.go [hunk 1]
```diff
@@ -27,15 +27,15 @@ func (bz *HexBytes) Unmarshal(data []byte) error {
 	return nil
 }
 
-// MarshalJSON implements the json.Marshaler interface. The hex bytes is a
-// quoted hexadecimal encoded string.
+// MarshalJSON implements the json.Marshaler interface.  The encoding is a JSON
```
Comment (cmwaters) [inline libs/bytes/bytes.go:30]: nit: extra space
```suggestion
// MarshalJSON implements the json.Marshaler interface. The encoding is a JSON
```

**Review by cmwaters — APPROVED** [review]
LGTM 👍 

Does this need any tests?

**Review by marbar3778 — APPROVED** [review]
love seeing this come through.

**Comment by creachadair** [comment]
> LGTM 👍
> 
> Does this need any tests?

There are already tests that do cover the existing behaviour, but if there are additional cases you think are worth adding, I will be glad to do so.

**Comment by creachadair** [comment]
Actually, there is one case the existing tests don't cover, but that some downstream tests do, viz., the expectation that digits will be capitalized. I'll add a case for that, which wasn't being exercised previously.

**Pull request merged**